manifests/ssltools/nssdb.pp

```diff
@@ -33,18 +33,13 @@
     owner  => 'root',
     group  => $group,
     mode   => '0640',
-  } ->
-  exec { 'create-nss-db':
-    command => "certutil -N -d '${nss_db_dir}' -f '${nss_db_password_file}'",
-    path    => '/usr/bin',
-    umask   => '0027',
-    group   => $group,
-    creates => $nssdb_files,
-    require => Package['nss-tools'],
-  } ->
-  file { $nssdb_files:
-    owner => 'root',
-    group => $group,
-    mode  => '0640',
+  }
+
+  nssdb { $nss_db_dir:
+    nssdb_files   => $nssdb_files,
```

Comment:
This should not be needed as this is an detail of the NSS database implementation.

Reply:
The `delete_nssdb` method in the provider also relies on this parameter currently.

Reply:
We can and should manage the owner/group/mode of the files inside the nssb directory but a user of the provider should not have to declare what those files are -- they are internal details, part of how nssdb works,  that a user should not have to care about. They are defined in the current implementation because there needed to be some way to make the creation idempotent. With the provider abstraction, you can do this more natively. For example, like https://github.com/theforeman/puppet-certs/pull/344#discussion_r640539099